### DISCUSSION TITLE
How can I create a custom RSS feed when using Statiq.dev (with Clean Blog theme).

### DISCUSSION THREAD

**Discussion by TechWatching** [statiqdev/Discussions#194] [Q&A] [answered]
In addition to the existing RSS feed, I need a custom one that will only show some articles with some specific tags (the aim is to have a RSS feed only about dotnet content). I was wondering how I can do that.

I have looked at the Feeds Pipeline in the source code, I have seen that the configuration comes from the feed.yml file of the theme, but I have trouble to see how to configure correctly the feeds to add a new feed in output with a specific filter (different from the other feeds).

Can someone explain to me what I need to do?

**Comment by daveaglick** [comment, marked as answer]
Have you seen this page in the guide: https://www.statiq.dev/guide/web/feeds? It includes all the various front-matter settings you can use to define what the feed should contain. In general, any YAML or JSON file that contains the frontmatter `FeedRss` as a value will be treated as a feed file. So to create additional feeds, create YAML files and add `FeedRss: true`, etc. to the top of them. Those files can go right in your own `input` folder and the output RSS file should have a similar path as the input YAML file (with the appropriate extension, of course).

For many of these values you might want to use [computed metadata values](https://www.statiq.dev/guide/documents-and-metadata/metadata-values#computed-values) like this:
```
FeedItemTitle: => doc.GetString("Title") + " In My Feed"
```

Is that enough to get started playing with feeds?

**Reply by TechWatching** [reply]
Yes I have seen this page but I realize I misunderstand it, I did not get that you could have many feed files. It much clearer thanks! And when you understand it, it's quite simple to use I must say.

One thing I struggle with however is to correctly debug the code doing the work. For instance, I noticed that my computed metadata value was filtering all the documents `FeedFilter: => GetBool("IsPost") && !GetBool("IsPostArchive") && GetList<string>("Tags").Contains("net")` but I have not been successful debugging the modules (FilterDocuments for instance) to see what metadata values were on my documents and why my filter was not working properly. Do you have advice on how to do that ?

I succeeded putting a breapoint in the FilterDocuments class (even if it's not in my project code), but then I can't evaluate the metadata values in a document, the debugger quits when I do that.

**Reply by daveaglick** [reply]
Yes! The magic trick for debugging is to use `--attach` on the command line. When you do that, Statiq will pause and print the process number, then you can find and attach to that process from any IDE.  Alternatively, since Statiq runs inside your own app, you should be able to start a debug session for that app right from the IDE too.

If you're thinking about debugging Statiq itself, that's what I mainly use the `--attach` option for. Basically you  clone the Statiq code (either Web or Framework depending on what you want to debug) then make sure your own app references that project file you cloned instead of the Statiq NuGet package. Then when you use `--attach` you can attach to the process from an IDE that's loaded the code you cloned.

Make sense?

**Reply by TechWatching** [reply]
In fact my problem is not really about attaching the debugger but about debugging the framework itself.

I wanted to know why my condition about the tag on the FeedFilter was failing (the feed is empty) so I tried to see what was happening in debug. Unfortunately, I am not able to do see the metadata of each document in debug: for instance in the FilterDocument.cs if I wan tot see the metadata in `context.Inputs` just before the documents are filtered I can't (the debugger says expanding will force the enumeration and fails when I try). Any idea how to debug that properly ?

**Reply by TechWatching** [reply]
I found out, what was the problem with my condition, nothing wrong,  I just get rid of the condition on archived posts are I don't mind having them in my RSS feed. Still, I would be interested to know what is the proper way to see the metadata of the documents while debugging.

**Comment by AlexHedley** [comment]
@TechWatching do you have a working example of this?

@daveaglick is there a way to have it dynamically filter by any tag?

**Reply by TechWatching** [reply]
Yes, you can find one here : https://github.com/TechWatching/techwatching.dev/blob/statiqdev/input/dotnet_feed.yml

**Reply by AlexHedley** [reply]
That’s great, thanks. Sorry for the follow up, I must be missing something obvious:

Main RSS Feed:
https://techwatching.dev/feed.rss

Just .NET?
- https://techwatching.dev/dotnet_feed.rss
- https://techwatching.dev/dotnet.rss
- https://techwatching.dev/posts/dotnet_feed.rss
- https://techwatching.dev/posts/dotnet.rss

**Reply by TechWatching** [reply]
That's because I migrated my website from Statiq to Nuxt Content a few months ago. I'm mostly using Vue.js for front development at work nowadays so I wanted to use that for my blog too. The code I showed you worked fine but is on the Statiq branch of my repo that contains the old code of my blog. I just forgot to migrate this specific feed with the rest of my blog

**Comment by AlexHedley** [comment]
@daveaglick I created a new file `src\input\dotnet_feed.yml`

```yaml
FeedSources: => GetString("PostSources")
FeedFilter: => GetBool("IsPost") && GetList<string>("Tags").Contains("dotnet")
FeedOrderKey: Published
FeedOrderDescending: true
FeedSize: 15
FeedRss: true
FeedAtom: true
```

When running `dotnet run -- preview` I get the following output

```bash
[INFO] -> Feeds/Process » Starting Feeds Process phase execution... (0 input document(s), 3 module(s))
[ERRO] Feeds/Process » ForEachDocument » ExecuteConfig » FilterDocuments » Value cannot be null. (Parameter 'source')
```

Is there anything else I'm missing in the above file or any other config I should add/update?

(Also what would the url be? The default one is http://localhost:5080/feed.rss)
